project/yuanhenglee/README.rst

```diff
@@ -0,0 +1,179 @@
+# BasiCPP Pitch: A C++ implementation for AMT(Automatic Music Transcription)
+
+BasiCPP Pitch is a instrument-agnostic and polyphonic capable AMT(Automatic
+Music Transcription) library written in C++.
+
+## Basic Information
+
+BasiCPP Pitch is an instrument-agnostic and polyphonic-capable AMT (Automatic
+Music Transcription) library written in C++.
+
+- github repository:[repo](https://github.com/yuanhenglee/basicpp-pitch)
+
+Provide any compatible audio file, the library will generate a MIDI file with
+the notes it detected. The library also provides Python API, which is
+implemented by pybind11, to make it easier to use. 
+
+The AMT model we used is from Spotify's
+[basic-pitch](https://github.com/spotify/basic-pitch). More information about
+the model can be found in the research paper, [A Lightweight Instrument-Agnostic
+Model for Polyphonic Note Transcription and Multipitch
+Estimation](https://arxiv.org/abs/2203.09893).
+
+## Problem to Solve
+
+In music information retrieval (MIR), Automatic Music Transcription (AMT) aims
+to convert raw audio recordings into symbolic representations like sheet music
+or MIDI files.
+
+One of the significant challenges in AMT is accurately transcribing polyphonic
+audio, where multiple notes are played simultaneously. A practical solution
+involves audio preprocessing techniques, e.g., Constant-Q Transform (CQT), to
+represent audio in the frequency domain. This specialized transform can capture
+the harmonic structure of music, which is vital for polyphonic transcription. By
+stacking harmonics, we create a comprehensive frequency representation for each
+time frame, enabling the subsequent steps to better discern individual notes in
+the presence of harmonically rich audio. We employ a Convolutional Neural
+Network (CNN) architecture to generate notes from the preprocessed audio frames. 
+
+## Prospective Users
+
+Music producers and engineers can employ it to transcribe audio recordings into
+MIDI files, facilitating tasks like remixing and editing. Music researchers and
+analysts can conduct in-depth compositional studies, while educators and
+students can use it for ear training and composition exercises.
+
+## System Architecture
+
+![image](https://github.com/yuanhenglee/basicpp-pitch/blob/master/pics/NSD_project_flowchart.drawio.png)
+
+### Input
+
+The system takes an audio file that needs to be transcribed. as input.
```

Comment:
How many analytical approaches do you want to implement in the project time frame?

Reply:
By analytical approaches, does that mean how many modules do I have to implement to accomplish the problem?
If that is the case, the primary two will be CQT and the inference module for the CNN.
I'm sorry if I misunderstood the question.